### PR TITLE
Update/module ibc docs

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at adding IBC documentation for accounts and modules. 
There are 3 parts to these docs : 
- Overview : Allows to understand what the feature allows. Good for application architects
- Integration : Allows to integrate with the feature. Good for application developers
- Maintenance: Allows to understand how the abstract framework functions. Good for abstract maintainers

For now I'm not using the SDK or Abstract-Client at all in those docs. We should find a spot for the SDK to intergate in or between those 3 parts 

### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/docs/mdbook-admonish.css
Comment (CyberHoward) [inline framework/docs/mdbook-admonish.css:1]: What happened here?

Comment (CyberHoward) [reply framework/docs/mdbook-admonish.css:1]: nvm, due to version bump :)

File: framework/docs/book.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ create-missing = false
 
 [preprocessor.admonish]
 command = "mdbook-admonish"
-assets_version = "2.0.1"    # do not edit: managed by `mdbook-admonish install`
+assets_version = "3.0.2" # do not edit: managed by `mdbook-admonish install`
```
Comment (CyberHoward) [inline framework/docs/book.toml:16]: We should also update version for this in CI, or it will fail.

Comment (CyberHoward) [reply framework/docs/book.toml:16]: done

**Review by cyberhoward — APPROVED** [review]

**Review by Buckram123 — APPROVED** [review]

**Comment by CyberHoward** [comment]
Great job on these docs btw @Kayanski . Only thing that I think we should add are some diagrams.

**Pull request merged**